PWGHF/TableProducer/derivedDataCreatorDstarToD0Pi.cxx

```diff
@@ -273,13 +283,19 @@ struct HfDerivedDataCreatorDstarToD0Pi {
         auto prong1 = candidate.template prong1_as<TracksWPid>();
         auto prongSoftPi = candidate.template prongPi_as<TracksWPid>();
         double y = candidate.y(o2::constants::physics::MassDStar);
-        double massDstar = candidate.invMassDstar();
         std::vector<float> mlScoresDstarToD0Pi;
-        bool isD0 = prongSoftPi.sign() < 0;
         if constexpr (isMl) {
           std::copy(candidate.mlProbDstarToD0Pi().begin(), candidate.mlProbDstarToD0Pi().end(), std::back_inserter(mlScoresDstarToD0Pi));
         }
-        fillTablesCandidate(candidate, prong0, prong1, prongSoftPi, isD0 ? 0 : 1, y, massDstar, flagMcRec, origin, mlScoresDstarToD0Pi);
+        if (candidate.signSoftPi() > 0) {
+          double massDstar = candidate.invMassDstar();
+          double invMassD0 = candidate.invMassD0();
+          fillTablesCandidate(candidate, prong0, prong1, prongSoftPi, 0, massDstar, invMassD0, y, flagMcRec, flagMcRecD0, origin, nTracksDecayed, ptBhadMotherPart, pdgBhadMotherPart, mlScoresDstarToD0Pi);
+        } else {
+          double massDstar = candidate.invMassAntiDstar();
+          double invMassD0 = candidate.invMassD0Bar();
+          fillTablesCandidate(candidate, prong1, prong0, prongSoftPi, 1, massDstar, invMassD0, y, flagMcRec, flagMcRecD0, origin, nTracksDecayed, ptBhadMotherPart, pdgBhadMotherPart, mlScoresDstarToD0Pi);
+        }
```

Comment:
Avoid duplicating the function call by keeping only the assignments inside the `if else` blocks.